src/parser/mod.rs

```diff
@@ -15018,6 +15018,9 @@ impl<'a> Parser<'a> {
 
     /// Parse a FETCH clause
     pub fn parse_fetch(&mut self) -> Result<Fetch, ParserError> {
+        if dialect_of!(self is SnowflakeDialect) {
+            return self.parse_snowflake_fetch();
+        }
```

Comment:
It looks like the snowflake behavior is mostly a subset of the other dialects? If that's the case I think a reasonable approach would be to make the parser logic for the existing `parse_fetch` more permissive and accept the optional components regardless of dialect